Refresh design, refactoring code for much easy templating and code reading

bouks opened the pull request:
**Enhancements :**

- new responsive design

- replacing png bars by css bars

- replacing png clocks by css clocks

- choice to have countries flags displayed with image or utf8 char
 
- replacing js tooltip system by css system

- add average filesize in pages report and downloads report

- code cleaning

**New functionnalities :**
 
- css vertical bars growing

- in-table horizontal bars integrated in each data columns (and with no extra column)

- table rows hovering

- new world map with hovering animation

- css vars for better and easy customisation

- stylesheet heritage or replacement

- Dark Mode

- Ratios table

- some expandable tables

- AWStats update notification (JSON::Parse modules needed to activate). Notification shows in the footer's page

**Updating considerations :**

**Plugins using "ShowInfoHost" and "ShowInfoURL"  MUST now return a string instead of directly printing (for reports)**


**CONF file changes:**

- new FlagsType option to choose image or utf countries flags

- new ShowMap option to show or not a worldmap in countries reporting

- new StylesheetMode option to choose if your stylesheet override awstats stylesheet or if awstats stylesheet is ignored

- removing BarHeight option _(replaced by a css var in stylesheet)_

CONF visual options are migrating to css (BarWidth, colors, ShowBars etc...)

**Close issues :**

- https://github.com/eldy/AWStats/issues/260

- https://github.com/eldy/AWStats/issues/258

- https://github.com/eldy/AWStats/issues/235

- https://github.com/eldy/AWStats/issues/233

Still WIP but usable.

![image](https://github.com/user-attachments/assets/d8e83050-79e0-4b00-9636-f30f6d427966)

More and updated screenshots at : https://github.com/eldy/AWStats/issues/233

bouks committed:
Remove images bars, replace by css only

Remove need of images for bar rendering. Using only css classes.

bouks committed:
set background color to css

Remove bgcolor, replace by newly color- css class introduced by css bars.

bouks commented:
On the second commit, title is not complete, should be : "set background TH color to css"

bouks committed:
Add color vars to default css, saving lines

bouks committed:
Reduce year size in month bars stats

Set "font-size: 9px"

-> same presentation as day of  month

bouks committed:
Change font definitions

Set font to sans-serif, system-ui
Remove duplicates font family
Change font size unit to rem unit

bouks committed:
Remove use of png for clock

Remove pngs, use of full css
clean css

bouks committed:
Fix clock centering

bouks committed:
Fix perl var in css files (removing)

IGLOU-EU commented:
Congrats, review all this old Perl is a nightmare :rofl: 
I like the "HtmlBar" function, much better.
Why are you keeping some embedded style like "<div style=\"font-size: 0.5rem;\">$YearRequired</div>" ?

bouks commented:
@IGLOU-EU
Thanks.
I'm not a perl coder, so it's like holidays from my point of view.  😄 

For the font-size, i was just converting from px to em while reading the code. But, sure, removing styles, old html attributes and set css classes is good.

We'll see. If @eldy merge the commits and is interested to move further this way (for the 8.0) i can do some rendering stuffs, opening for much easier theming and more modern design. It's not a complex work, it takes just some time.

bouks commented:
css clocks. clean. :)
![image](https://github.com/user-attachments/assets/e5658f3f-0eea-4174-a20e-3cdbee5ef120)

eldy commented on wwwroot/css/awstats_bw.css:
```diff
@@ -1,31 +1,50 @@
-.aws_title  { font: 13px verdana, arial, helvetica, sans-serif; font-weight: bold; background-color: #BBBBBB; text-align: center; margin-bottom: 0; padding: 1px 1px 1px 1px; }
-.aws_blank { font: 13px verdana, arial, helvetica, sans-serif; background-color: #FFFFFF; text-align: center; margin-bottom: 0; padding: 1px 1px 1px 1px; }
+body { body { font: 0.75rem sans-serif, system-ui; background-color: #FFFFFF; margin-top: 0; margin-bottom: 0; } }
+.aws_bodyl { background-image: url(/awstatsicons/other/backleft.png); background-repeat: repeat-y; }
```
May be we should use $DirIcons here ?

bouks replied:
Hi @eldy
It's a copypaste from awstats_default. I don't know the difference between awstats_default.css and awstats_bw.css, and their use except for example, because css is hardcoded in awstats.pl. Maybe we can keep only one.
By the way, did you hardcoded the css for a special purpose ? Or maybe we can include a default "$StyleSheet" from css folder and remove the hardcoded one ?

For the backleft image, maybe we just could remove this image. And the button.gif also.

Another consider i have is to refactor all the calculations for tables. For example, in HTMLMainMonthly, these are written 3 times, and calculated 2 times (or more if there are multiples plugins). We could just use the ShowGraph plugin calculated vars and then dispatch them for graph plugins and rendering bars table and data table.

bouks committed:
Refactoring & redesigning month stats data table

Refactoring rendering of header, cells and footer (usable for other tables).

Add some css classes to the html tags.

Bars are integrated to the table (based on $BarWidth, =>100 is cool).

bouks committed:
Update Days of month table (and month)

Using refactoring made with month table.

Set average footer.

Fix averages to calculate only with not empty days.

Set averages for Month table.

bouks committed:
Update Days of week table

See previous commit.

bouks committed:
Update hours table

Like previous commits.
+ set flexible

bouks committed:
Fix Byte Format

bouks committed:
Remove the need of country flag pngs

In domain tables.

Using UTF8 characters instead.

bouks committed:
clocks night and day

bouks committed:
Adding effects on lines hovering

bouks committed:
activate U and V stats to domains/country table

add options to ShowDomainsStats and defaulted to "UVPHB''.

Already Coded by Josep Ruano here :
https://github.com/eldy/AWStats/blob/269e9736ce02e65c25a2a4ad3e5be921ea92b878/wwwroot/cgi-bin/awstats.pl#L13217-L13243

bouks committed:
change translations

- "différents" > "uniques" (plus court, évite les sauts de lignes)

- "Heures" > "Heure"

bouks committed:
Update Countries table

- global update of redesigning, refactoring tasks

- reintroducing a world map

bouks committed:
Introduce img folder (worldmap)

- Add folder in installation setup
- Update documentation

bouks committed:
Cleaning Days of Month process

- rearrange code : declarations => datas manipulations => rendering
- avoid loops : days are now parsed only once

bouks committed:
Fix countries

resolve the width problems.
set flexible, new design.

bouks committed:
Fix bars

the 'bk' problem :)

bouks committed:
Fix space/maj

bouks committed:
Removing the wrapping table of countries

bouks committed:
Cleaning Months process

- rearrange code : declarations => datas manipulations => rendering
- avoid loops : days are now parsed only twice

- Fix also average for days of month table.

bouks committed:
Cleaning html headers generator ! "AddHTMLStyles_" PLUGINS : MUST READ COMMENT !

- Remove backslashes from strings
- Cleaning code

- Set a proper function to render css rules

##### NOTE #####
"AddHTMLStyles_" plugins MUST now RETURN A STRING instead of printing freely to the output.
It permits future validation methods (security) and easy refactoring for awstats.pl development.
################

- Start works for weekdays table

bouks commented:
@eldy 

Important modification regarding using _**"AddHTMLStyles_$pluginname"**_ plugins in this commit : https://github.com/eldy/AWStats/pull/256/commits/5d41ad7364a5edc8548c774ca178af879c194a50

Read the comment.

bouks committed:
default BarHeight to 75 + borders

- remove borders

bouks committed:
reset heures au pluriel

bouks committed:
Introduce ShowBars option

To see or not the tables with vertical bars
Default : ShowBar = 1

bouks committed:
add class to average row

bouks committed:
Set some css vars

- converting perl color vars in css vars
- set a class for header domain name
- remove label for domain name
- higher font size and bolding for domain name
- remove default centered text alignment  in tables cell

bouks committed:
Refactoring bar system

bouks committed:
Remove unnecessary quotes

and replace double quotes by simple quotes.

bouks committed:
Clean HTMLMainDaysofWeek

refactoring, clean code

bouks commented:
Note :

The worldmap used to show countries is SVG.

Github count it as code and add 3272 lines to the PR.

bouks committed:
css - fluidify effect on hovering

bouks committed:
Add $StyleSheetMode option

When templating with a personnal stylesheet, the user can choose to only override the built-in css (included in the awstats.pl file), or totally ignore the built-in css.

bouks committed:
verify month (DateIsValid)

bouks committed:
active ShowBars on weekday

bouks committed:
add width option to bars

bouks committed:
js+css - country effect on hovering

bouks committed:
js refactoring

bouks committed:
HTML cleaning

remove obsolete attributes

bouks committed:
Update update message

bouks committed:
clean site header

bouks committed:
reset  domains/country on white bg

bouks committed:
flex container

bouks committed:
Lighten default pages stats blue background

bouks committed:
update fr translation

bouks committed:
new headers

- for global page
- for sections

+ css stuffs

bouks committed:
Remove parenthesis on ratio summary

bouks committed:
Fix some css

- container width
- center bars tables
- make the world go round :)

bouks committed:
css fix unboldy things

bouks committed:
visuals

bouks committed:
update fr translation

bouks committed:
redesign header + summary

bouks committed:
Merge branch 'eldy:develop' into develop

bouks committed:
cleaning code HTMLMainHosts

bouks committed:
add fr msg 179 > 188

bouks committed:
downloads table

bouks committed:
insert data-tables

bouks committed:
reduce "links" size

bouks committed:
Bars grow on hover

+ some css work to be more responsive
+ optimize space, reduce header and tables heights
+ moving HTMLMainDownloads to be in order of table on the page (practical if you must repeat things)

bouks committed:
re-active extra_head_html & tooltips

on tab_head.
move analysis date to save height.

bouks committed:
New tooltip system, plugin updates

without javascript

bouks committed:
Correct  functions comments

bouks committed:
finalizing tab_head  and tab_end

make them printable

bouks committed:
fix worldmap size

bouks committed:
adjust tooltip place

bouks committed:
renew tables on detailed pages

+ standardize email reports

bouks committed:
Fix table SMTP error codes

bouks commented:
Resume :

- new responsive design

- css vertical bars growing (removal of png) with optional showing

- in-table bars

- css clocks (removal of png)
 
- animation and hovering table rows
 
- new world map with animation

- new tooltip system

- stylesheet heritage or replacement

Soon :

- stylesheet example with css vars replacing awstats.conf vars (BarHeight, BarWidth and colors)

bouks committed:
Adjust clock size on bars

bouks committed:
Remove ShowBars option / remove BarWidth from bar calculation

bouks committed:
remove conf color var from css

bouks committed:
Set a navigation menu

bouks committed:
Remove ShowBars from conf file

bouks committed:
Remove BarHeight from conf

bouks committed:
Main Countries add full list to main page

with collapsed row for the not top and  a button to show them.
Fix countries not returning to normal state after unhovering.
Add one color to the map.
gray = no visit
brown = visit
orange = topvisit
yellow = hovering

bouks committed:
Add a ratio stats table

bouks committed:
rearrange tables for fluid navigation

bouks committed:
gain height

bouks committed:
add comment

bouks committed:
Change wrong word

bouks committed:
Comply screen size table to html template

bouks committed:
Comply robots table to html template

bouks committed:
comply host table

bouks committed:
Comply worms table

bouks committed:
Remove "ucfirst" messages from lang pack

replacing by lowercase version with  ucfirst perl function in awstats.pl

bouks committed:
conform session table

bouks committed:
Simplify screensize table

bouks committed:
update fr

bouks committed:
conform  file types table

bouks committed:
update file types labels

bouks committed:
Dark mode implemented + link fix in menu

bouks committed:
fix weekend color, remove  2nd line on bars

bouks committed:
return string on mainmonthly, dailys tables, countries, hours table

bouks committed:
visually reorganizing tables

bouks committed:
Fix clock not turning

bouks committed:
fix ratio columns, move sections

bouks committed:
if no data, don't show ratio

bouks committed:
set font more fluid

bouks committed:
if not worms return nothing

bouks committed:
Starting HTMLMainPages cleanup

Need to alter HTMLShowURLInfo (return instead of prints) to go further.

bouks committed:
Refactor HTMLShowURLInfo

don't print return string + code cleaning

bouks commented:
Enhancements :
- new responsive design

- css vertical bars growing (removal of png) with optional showing

- in-table bars

- css clocks (removal of png)
 
- animation and hovering table rows

- new tooltip system

- code cleaning

New functionnalities :
 
- new world map with animation

- css vars for better and easy customisation

- stylesheet heritage or replacement

- Dark Mode

- Ratios table

Still WIP but usable.

bouks committed:
css : grow on portrait and 2dppx

bouks committed:
proper filter ShowLinksOnUrl

bouks committed:
Refactor "details" page's link build on top of table.

bouks committed:
clean code

bouks committed:
confom pages-url table

bouks committed:
AWStats update notification (in page footer)

bouks commented on wwwroot/cgi-bin/awstats.pl:
Needed JSON::Parse to show updates.
If not available, just show nothing.

bouks committed:
conform Downloads table

bouks committed:
TODO add devicePixelRatio

bouks committed:
cleanup

bouks committed:
conform mainOS table

bouks committed:
update some browser's icon

bouks committed:
conform browsers table

bouks committed:
conform keyphrases keywords tables

bouks committed:
Change some visual rendering

bouks committed:
conform http status table

bouks committed:
conform referers table

bouks committed:
Countries expandable button

bouks committed:
alldomains page

bouks committed:
HTMLShowDomains return instead of print

bouks committed:
Don't print form, update button and nav

bouks committed:
print styles and sticky map on alldomains page

bouks committed:
urls pages (all, entries, exits)

bouks commented:
actually saved 1115 lines (5%) in awstats.pl 😃

bouks committed:
downloads page

bouks committed:
Mainhost + ShowHostInfo -> return instead of print

bouks committed:
Fix clock icon retarding 1 hour

bouks committed:
(css) Fix data was uppercase on countries

bouks committed:
(css) Dark mode more darker

bouks committed:
section help button (tooltip)

bouks committed:
styling section links

bouks committed:
shorten "Taille moyenne" > "Poids moyen"

bouks committed:
pages-url table - add data (bandwidth) column and set bars for "others" line

bouks committed:
plugins geoip/geoip_city_maxmind returning html string

+ th not bg background
+ "remove" some tooltip (old system) waiting to refactor

bouks committed:
Add links to free geoip databases

bouks committed:
Add option for flags ( image | utf8). default : 'image"

bouks committed:
Add conf option to show worldmap or not

bouks committed:
(js) Don't change worldmap's top in detail page if showmap = 0

bouks committed:
"Add" Message[56] -> '-' for "no result" (geoip_city_maxmind, geoip)

bouks committed:
geoip_city_maxmind, geoip, return only content

The cell itself is build by awstats in sub HTMLShowHostInfo

Set background to plugins columns

Arrange paddings

bouks committed:
(geo) change functions doc , remove span

bouks committed:
css gui

"top 10" smaller
misc table at bottom
full width tables on portrait mode
style adjustments

bouks committed:
Update Geo plugins ShowInfoHost subroutine

returning only content
cleaning code

bouks committed:
Conform hosts detail pages + some general colors and css

bouks committed:
Fix title url details and remove sub HTMLShowURLDetail

bouks committed:
Remove sub HTMLShowDownloads

bouks committed:
Remove sub HTMLShowDomains

bouks committed:
Conform robots details pages and remove sub HTMLShowRobots

bouks committed:
Change Format_Byte font size depending B, KB, MB...

bouks committed:
(css) mod bytes sizes + mod hover anim

bouks committed:
Set html5 doctype and utf-8 charset + some css

eldy commented:
I tested your PR,but this is what i got with the unit test
![image](https://github.com/user-attachments/assets/d1f06fb8-3e51-4301-9b38-e25160a0c785)

Graph are no more visible.
The summary part does not show any more difference between human and not human traffic
Also it is difficult to see separation between each array.
I have also some html error in some html part.
It is difficult to me to report what's wrong on each change because all changes are cumulated into the same PR making not possible for me to analyse each change and validate them one by one.

bouks commented:
Hi @eldy 

I didn't work on the graph for the moment because i don't use it and it is noted as experimental in the conf file. I though it is "on development", and i saw in issues that google one doesn't work anymore.

Sorry. It seems i don't understand very well the pr system from a fork. I did a pr for the bars and the clock, and after all my commits added to the PR.

PS: i managed to find the problem with the graphapplet plugin (printing directly, and format_byte function i included html in). I fixed it, but i can't test since i have not java in all my browsers (but all is well written in the html code with the fix). I think remember java applets in browser have been discontinued for some years.

ShaiMagal commented:
"Ofttopic" - problem with charts is already discussed here #249 (not solved, there is necessary to change "chart library")

dannybeckett commented:
UTF8 country flags is a great idea. They don't display on Windows very well, but it is very easy to fix using https://github.com/talkjs/country-flag-emoji-polyfill
